compiler_gym/datasets/datasets.py

```diff
@@ -0,0 +1,480 @@
+            self._active_datasets.remove(dataset_name)
+        return deactivated
+
+    def install(self, dataset: Union[str, Dataset]) -> bool:
+        """Install a dataset.
+
+        Run the local installation procedure for the given dataset. This will
+        perform one-time setup of the dataset such as downloading remote
+        resources. See :meth:`Dataset.install()
+        <compiler_gym.datasets.Dataset.install>`.
+
+        Installing a dataset does not make it inactive if it is not already.
+
+        :param dataset: A :meth:`Dataset <compiler_gym.datasets.Dataset>`
+            instance, or the name of a dataset.
+
+        :return: :code:`True` if the dataset was installed, :code:`False` if
+            it was already installed.
+        """
+        dataset: Dataset = self.dataset(dataset)
+        installed = not dataset.installed
+        dataset.install()
+        return installed
+
+    def uninstall(self, dataset: Union[str, Dataset]) -> bool:
+        """Uninstall a dataset.
+
+        Run the local uninstall procedure for the given dataset. This will
+        perform one-time clean up of the dataset such as removing locally cached
+        files. See :meth:`Dataset.uninstall()
+        <compiler_gym.datasets.Dataset.uninstall>`.
+
+        Uninstalling a dataset does not make it inactive, and does not remove it
+        from this datasets collection.
+
+        :param dataset: A :meth:`Dataset <compiler_gym.datasets.Dataset>`
+            instance, or the name of a dataset.
+
+        :return: :code:`True` if the dataset was uninstalled, :code:`False` if
+            it was already uninstalled.
+        """
+        dataset: Dataset = self.dataset(dataset)
+        uninstalled = dataset.installed
+        dataset.uninstall()
+        return uninstalled
+
+    def require(self, dataset: Union[str, Dataset]) -> bool:
+        """Require that a dataset is available for immediate use.
+
+        This performs any one-time setup of the dataset so that it can be used.
+        It is equivalent to:
+
+            >>> env.datasets.add(dataset)
+            >>> env.datasets.activate(dataset)
+            >>> env.datasets.install(dataset)
+
+        :param dataset: A :meth:`Dataset <compiler_gym.datasets.Dataset>`
+            instance, or the name of a dataset.
+
+        :return: :code:`True` if the dataset was made available, :code:`False`
+            if it was already available.
+        """
+        added = False
+        # Use an attribute check rather than isinstance() to allow mocking the
+        # Dataset class in testing.
+        if hasattr(dataset, "name"):
+            added = self.add(dataset)
+            dataset_name = dataset.name
+        else:
+            dataset_name = resolve_uri_protocol(dataset)
+        activated = self.activate(dataset_name)
+        installed = self.install(dataset_name)
+        return added or activated or installed
+
+    def benchmarks(self) -> Iterable[Benchmark]:
```

Comment:
Presumably datasets is finite?
If the first dataset is infinite, then you will never get any benchmarks from the second.
Consider:
```
iters = [d.benchmarks().iterator() for d in self.datasets()]
while iters:
  it = iters.pop_front()
  if it.has_next(): # I don't know how Python does iterators
    yield it.next()
    iters.push_back(it)
```
That way they get a benchmark from each suite until exhaustion and an infinite suite can't hog the limelight.

(There are better ways to achieve what I wrote above).

Reply:
I like this a lot!

Python iterators raise `StopIteration` when empty, so its:

```py
        iters = deque((d.benchmarks() for d in self.datasets()))
        while iters:
            it = iters.popleft()
            try:
                yield next(it)
                iters.append(it)
            except StopIteration:
                pass
```

Reply:
Implemented

Reply:
Should this also have an `include_deprecated` flag?

Reply:
I don't know if its an important use case, but sure why not :) done